pkg/arvo/app/btc-node-hook.hoon

```diff
@@ -0,0 +1,373 @@
+      :: %unload-wallet
+      :: %wallet-create-fundedpsbt
+      :: %wallet-lock
+      :: %wallet-passphrase
+      :: %wallet-passphrase-change
+      :: %wallet-process-psbt
+  ==
+::
+++  btc-node-store-poke
+  |=  [pax=path act=btc-node-store-action]
+  ^-  card
+  :*  %pass   pax
+      %agent  [our.bowl %btc-node-store]
+      %poke   [%btc-node-store-action !>(act)]
+  ==
+::
+++  default-wallet
+  .^  @t
+      %gx
+      (scot %p our.bowl)
+      %btc-node-store
+      (scot %da now.bowl)
+      /default-wallet/noun
+  ==
+::
+++  n-wallets
+  .^  @ud
+    %gx
+    (scot %p our.bowl)
+    %btc-node-store
+    (scot %da now.bowl)
+    /n-wallets/noun
+  ==
+::
+++  endpoint-url
+  |=  [act=btc-node-hook-action]
+  ?.  ?|  ?=([%abandon-transaction *] act)
+          ?=([%abort-rescan *] act)
+          ?=([%add-multisig-address *] act)
+          ?=([%backup-wallet *] act)
+          ?=([%bump-fee *] act)
+          ?=([%dump-privkey *] act)
+          ?=([%dump-wallet *] act)
+          ?=([%encrypt-wallet *] act)
+          ?=([%fund-raw-transaction *] act)
+          ?=([%get-balance *] act)
+          ?=([%get-addresses-by-label *] act)
+          ?=([%get-address-info *] act)
+          ?=([%get-new-address *] act)
+          ?=([%get-raw-change-address *] act)
+          ?=([%get-received-by-address *] act)
+          ?=([%get-received-by-label *] act)
+          ?=([%get-transaction *] act)
+          ?=([%get-unconfirmed-balance *] act)
+          ?=([%get-wallet-info *] act)
+          ?=([%import-address *] act)
+          ?=([%import-multi *] act)
+          ?=([%import-privkey *] act)
+          ?=([%import-pruned-funds *] act)
+          ?=([%import-pubkey *] act)
+          ?=([%import-wallet *] act)
+          ?=([%key-pool-refill *] act)
+          ?=([%list-address-groupings *] act)
+          ?=([%list-labels *] act)
+          ?=([%list-lock-unspent *] act)
+          ?=([%list-received-by-address *] act)
+          ?=([%list-received-by-label *] act)
+          ?=([%lists-in-ceblock *] act)
+          ?=([%list-transactions *] act)
+          ?=([%list-unspent *] act)
+          ?=([%lock-unspent *] act)
+          ?=([%remove-pruned-funds *] act)
+          ?=([%rescan-blockchain *] act)
+          ?=([%send-many *] act)
+          ?=([%send-to-address *] act)
+          ?=([%set-hd-seed *] act)
+          ?=([%set-label *] act)
+          ?=([%set-tx-fee *] act)
+          ?=([%sign-message *] act)
+          ?=([%sign-raw-transaction-with-wallet *] act)
+          ?=([%wallet-create-fundedpsbt *] act)
+          ?=([%wallet-lock *] act)
+          ?=([%wallet-passphrase *] act)
+          ?=([%wallet-passphrase-change *] act)
+          ?=([%wallet-process-psbt *] act)
+      ==
+    endpoint
+  ?:  ?=([?(%dump-wallet %import-wallet) filename=@t] act)
+    (crip "{(trip endpoint)}wallet/{(trip filename.act)}")
+  ::  FIXME: fails when default-wallet is '' and more than 1 wallet
+  ::  has been created
```

Comment:
Considering the default `default-wallet` is `''`, this seems pretty common. Probably want to fix this prior to merge?

I don't understand why we have the empty name as default, anyway. Is that mirrored on the node side? I'm not very surprised it's giving issues with `wallet/` as top-level endpoint vs `wallet/` as pointing to the wallet with the empty name. (If I'm interpreting this situation correctly.)

Reply:
The full node creates a wallet called  `''`, so when only that wallet exists we can just use the url without specifying the wallet name. If we create another wallet, then we need to append `/wallet/<wallet_name>`. If we want to address the `''` wallet, the curl example here works `('http://127.0.0.1:18443/wallet/'`) but I couldn't make this work when creating a `request:http` (I get a `404` with no other message). I had assumed that I was not creating the `request:http` the right way, or that there was another way to do it because the curl works:
```
curl --location --request POST 'http://127.0.0.1:18443/wallet/' \       
--header 'Content-Type: application/json' \
--header 'Authorization: Basic dXJiaXRjb2luZXI6dXJiaXRjb2luZXI=' \
--data-raw '{"id":"curltext","method":"listtransactions","params":[]}'
```
but the `request:http` does not:
```
[ method=%'POST'
  url='http://127.0.0.1:18443/wallet/'
  header-list=~[[key='Content-Type' value='application/json'] [key='Authorization' value='Basic dXJiaXRjb2luZXI6dXJiaXRjb2luZXI']]
  body=[~ [p=82 q='{"params":[],"id":"list-transactions","jsonrpc":"0.2","method":"listtransactions"}']]
]
```

Reply:
Correction: This is not a problem when different wallets are created. This issue comes when multiple wallets are loaded at the same time. This is a bit more complicated then since wallets can be loaded/unloaded without the gall app knowing about it. If we just had that sync feature...